Library/Homebrew/os/mac/xcode.rb

```diff
@@ -225,16 +225,36 @@ def version
       def detect_version
         # CLT isn't a distinct entity pre-4.3, and pkgutil doesn't exist
         # at all on Tiger, so just count it as installed if Xcode is installed
-        return MacOS::Xcode.version if MacOS::Xcode.installed? && MacOS::Xcode.version < "3.0"
+        return MacOS::Xcode.version if MacOS::Xcode.installed? && XcodeVersion.is_lower(MacOS::Xcode.version, than: "3.0")
 
         [MAVERICKS_PKG_ID, MAVERICKS_NEW_PKG_ID, STANDALONE_PKG_ID, FROM_XCODE_PKG_ID].find do |id|
-          if MacOS.version >= :mavericks
+          if XcodeVersion.is_higher_or_equal(MacOS.version, than: :mavericks.to_s)
             next unless File.exist?("#{MAVERICKS_PKG_PATH}/usr/bin/clang")
           end
           version = MacOS.pkgutil_info(id)[/version: (.+)$/, 1]
           return version if version
         end
       end
     end
+
+    module XcodeVersion
```

Comment:
Make this a class that inherits from `Version` instead and override `<=>`. Also, `Xcode::Version` is probably a preferable name. Thanks!